src/Mvc/Mvc.Core/src/Builder/OpenApiEndpointConventionBuilderExtensions.cs

```diff
@@ -125,7 +125,52 @@ public static MinimalActionEndpointConventionBuilder ExcludeFromDescription(this
             return Produces<HttpValidationProblemDetails>(builder, statusCode, contentType);
         }
 
+#pragma warning disable CS0419 // Ambiguous reference in cref attribute
+        /// <summary>
+        /// Adds the <see cref="Accepts"/> to <see cref="EndpointBuilder.Metadata"/> for all builders
+        /// produced by <paramref name="builder"/>.
+        /// </summary>
+        /// <typeparam name="TRequest">The type of the request.</typeparam>
+        /// <param name="builder">The <see cref="MinimalActionEndpointConventionBuilder"/>.</param>
+        /// <param name="contentType">The request content type. Defaults to "application/json" if empty.</param>
+        /// <param name="additionalContentTypes">Additional response content types the endpoint produces for the supplied status code.</param>
+        /// <returns>A <see cref="MinimalActionEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
+        public static MinimalActionEndpointConventionBuilder Accepts<TRequest>(this MinimalActionEndpointConventionBuilder builder, string? contentType = null, params string[] additionalContentTypes)
+#pragma warning restore CS0419 // Ambiguous reference in cref attribute
+#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
+        {
+            Accepts(builder, typeof(TRequest), contentType, additionalContentTypes);
+
+            return builder;
+        }
+
+
+
+#pragma warning disable CS0419 // Ambiguous reference in cref attribute
+        /// <summary>
+        /// Adds the <see cref="Accepts"/> to <see cref="EndpointBuilder.Metadata"/> for all builders
+        /// produced by <paramref name="builder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="MinimalActionEndpointConventionBuilder"/>.</param>
+        /// <param name="requestType">The type of the request. Defaults to null.</param>
+        /// <param name="contentType">The response content type that the endpoint accepts.</param>
+        /// <param name="additionalContentTypes">Additional response content types the endpoint accepts</param>
+        /// <returns>A <see cref="MinimalActionEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
+        public static MinimalActionEndpointConventionBuilder Accepts(this MinimalActionEndpointConventionBuilder builder, Type requestType ,
+#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
+#pragma warning restore CS0419 // Ambiguous reference in cref attribute
+            string? contentType = null, params string[] additionalContentTypes)
```

Comment:
```suggestion
        public static MinimalActionEndpointConventionBuilder Accepts(this MinimalActionEndpointConventionBuilder builder, Type requestType , string? contentType = null, params string[] additionalContentTypes)
#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
```

We should figure out the cref warning.

Reply:
The overloads here are a bit problematic - 

```
Accepts(Type, string?, params string[]);
Accepts(string, params string[]);
```

You usually don't add new parameters to the left. Can we make the 2nd one require an array and be done with it e.g. 
`Accepts(IEnumerable<string> contentTypes, Type requestType);`